From: Yu Zhao <yuzhao@google.com>
Subject: mm-mglru-rework-workingset-protection-fix
Date: Tue, 24 Dec 2024 12:04:44 -0700

Revert change to folio_mark_accessed() that unncessarily dirties cache
lines shared between different cores.

Link: https://lkml.kernel.org/r/Z2sFzKmHPI0kI_fq@google.com
Signed-off-by: Yu Zhao <yuzhao@google.com>
Tested-by: kernel test robot <oliver.sang@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/swap.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/mm/swap.c~mm-mglru-rework-workingset-protection-fix
+++ a/mm/swap.c
@@ -395,7 +395,8 @@ static void lru_gen_inc_refs(struct foli
 
 	do {
 		if ((old_flags & LRU_REFS_MASK) == LRU_REFS_MASK) {
-			folio_set_workingset(folio);
+			if (!folio_test_workingset(folio))
+				folio_set_workingset(folio);
 			return;
 		}
 
_
